### ISSUE TITLE
MLP generalization

### ISSUE THREAD

**Issue by davidpitl** [umachkaalex/Neural-Networks#4]
Could it be generalised to work with neural networks that have n inputs and m outputs?

For example for this kind of model:

model.add(Dense(512, activation='relu', input_shape=(784,)))
model.add(Dropout(0.2))
model.add(Dense(512, activation='relu'))
model.add(Dropout(0.2))
model.add(Dense(num_classes, activation='softmax'))
